src/libraries/System.Security.Cryptography.Cose/tests/CoseHeaderMapTests.cs

```diff
@@ -0,0 +1,248 @@
+            // tstr / uint
+            Assert.Throws<InvalidOperationException>(() => map.SetValue(new CoseHeaderLabel(KnownHeaders.ContentType), -1));
+            // bstr
+            Assert.Throws<InvalidOperationException>(() => map.SetValue(new CoseHeaderLabel(KnownHeaders.Kid), "foo"));
+            // bstr
+            Assert.Throws<InvalidOperationException>(() => map.SetValue(new CoseHeaderLabel(KnownHeaders.IV), "foo"));
+            // bstr
+            Assert.Throws<InvalidOperationException>(() => map.SetValue(new CoseHeaderLabel(KnownHeaders.PartialIV), "foo"));
+        }
+
+        [Fact]
+        public void SetEncodedValue_KnownHeaders_ThrowIf_IncorrectValue()
+        {
+            var writer = new CborWriter();
+            writer.WriteNull();
+            ReadOnlyMemory<byte> encodedNullValue = writer.Encode();
+
+            var map = new CoseHeaderMap();
+            // only accepts int or tstr
+            Assert.Throws<InvalidOperationException>(() => map.SetEncodedValue(new CoseHeaderLabel(KnownHeaders.Alg), encodedNullValue));
+            // [ +label ] (non-empty array)
+            Assert.Throws<NotSupportedException>(() => map.SetEncodedValue(new CoseHeaderLabel(KnownHeaders.Crit), encodedNullValue));
+            // tstr / uint
+            Assert.Throws<InvalidOperationException>(() => map.SetEncodedValue(new CoseHeaderLabel(KnownHeaders.ContentType), encodedNullValue));
+            // bstr
+            Assert.Throws<InvalidOperationException>(() => map.SetEncodedValue(new CoseHeaderLabel(KnownHeaders.Kid), encodedNullValue));
+            // bstr
+            Assert.Throws<InvalidOperationException>(() => map.SetEncodedValue(new CoseHeaderLabel(KnownHeaders.IV), encodedNullValue));
+            // bstr
+            Assert.Throws<InvalidOperationException>(() => map.SetEncodedValue(new CoseHeaderLabel(KnownHeaders.PartialIV), encodedNullValue));
+        }
+
+        [Fact]
+        public void Enumerate()
+        {
+            var alg = new CoseHeaderLabel(KnownHeaders.Alg);
+            var crit = new CoseHeaderLabel(KnownHeaders.Crit);
+            var contentType = new CoseHeaderLabel(KnownHeaders.ContentType);
+            var kid = new CoseHeaderLabel(KnownHeaders.Kid);
+            var iV = new CoseHeaderLabel(KnownHeaders.IV);
+            var partialIV = new CoseHeaderLabel(KnownHeaders.PartialIV);
+
+            var map = new CoseHeaderMap();
+            map.SetValue(alg, (int)ECDsaAlgorithm.ES256);
+            //map.SetEncodedValue(crit, GetDummyCritHeaderValue());
+            map.SetValue(contentType, ContentTypeDummyValue);
+            map.SetValue(kid, s_SampleContent);
+            map.SetValue(iV, ReadOnlySpan<byte>.Empty);
+            map.SetValue(partialIV, ReadOnlySpan<byte>.Empty);
+
+            var writer = new CborWriter();
+            int currentHeader = KnownHeaders.Alg;
+            foreach ((CoseHeaderLabel Label, ReadOnlyMemory<byte> EncodedValue) in map)
```

Comment:
As this is a deconstruction, not a tuple declaration, it should be lowercase names.

```suggestion
            foreach ((CoseHeaderLabel label, ReadOnlyMemory<byte> encodedValue) in map)
```